src/compiler/checker.ts

```diff
@@ -13929,6 +13929,17 @@ namespace ts {
                             Diagnostics.The_expected_type_comes_from_the_return_type_of_this_signature,
                         ));
                     }
+                    if ((getFunctionFlags(node) & FunctionFlags.Async) === 0
+                        // exclude cases where source itself is promisy - this way we don't make a suggestion when relating
+                        // an IPromise and a Promise that are slightly different
+                        && !getTypeOfPropertyOfType(sourceReturn, "then" as __String)
+                        && checkTypeRelatedTo(createPromiseType(sourceReturn), targetReturn, relation, /*errorNode*/ undefined)

```

Comment:
I would have guessed it would be cheaper to unwrap `targetReturn`, so we wouldn't have to instantiate `Promise<source>`.

Reply:
`target` could be any number of compatible promise-likes which don't unwrap cleanly. It's much easier to wrap source. (And better mirrors what the runtime does)